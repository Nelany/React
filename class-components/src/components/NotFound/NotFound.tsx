import { useNavigate } from '@remix-run/react';
import classNames from 'classnames';
import Cookies from 'js-cookie';
import { useEffect, useState } from 'react';

const NotFound = () => {
  const navigate = useNavigate();
  const [theme, setTheme] = useState('light');

  useEffect(() => {
    const cookieTheme = Cookies.get('theme') || 'light';
    setTheme(cookieTheme);
  }, []);

  const containerClasses = classNames('error-content-container', theme);
  const buttonClasses = classNames('go-to-main-button', theme);

  const handleClick = () => {
    navigate(`/`);
  };

  return (
    <div className={containerClasses}>
      <h1 className="error-content">Ooops... Page not found!</h1>
      <img className="error-img" src="/cat.png" alt="Cat" />

      <button className={buttonClasses} onClick={handleClick}>
        Go to Main
      </button>
    </div>
  );
};

export default NotFound;
