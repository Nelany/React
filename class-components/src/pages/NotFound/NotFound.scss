@import '../../utils/variables.scss';

.error-content-container {
  width: 98vw;
  min-height: 98vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 30px;
  box-sizing: border-box;
}

.error-content-container.dark {
  background-color: $black;
  color: $green;

  .error-boundary-h3 {
    color: $green;
  }
}

.error-content {
  height: 60px;
  display: inline-block;
  margin: 50px;

}

.error-img {
  max-width: 70%;
  max-height: 40vh;
}

.error-boundary-h3 {
  margin-top: 30px;
}