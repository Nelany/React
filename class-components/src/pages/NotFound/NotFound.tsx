import { GoToMainButton } from '../../components/GoToMainButton/GoToMainButton';
import { useTheme } from '../../hooks/useTheme';
import './NotFound.scss';

export const NotFound = () => {
  const { theme } = useTheme();

  return (
    <div className={`error-content-container ${theme}`}>
      <h1 className="error-content">Ooops... Page not found!</h1>
      <img className="error-img" src="/cat.png" alt="Cat" />
      <GoToMainButton />
    </div>
  );
};
