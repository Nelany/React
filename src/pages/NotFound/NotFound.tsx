import { GoToMainButton } from '../../components/GoToMainButton/GoToMainButton';
import './NotFound.scss';

export const NotFound = () => {
  return (
    <div className="error-content-container">
      <h1 className="error-content">Ooops... Page not found!</h1>
      <img className="error-img" src="/owl.png" alt="Owl" />
      <GoToMainButton />
    </div>
  );
};
