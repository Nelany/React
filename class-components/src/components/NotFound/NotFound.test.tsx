import { render } from '@testing-library/react';
import { vi } from 'vitest';
import NotFound from './NotFound';

vi.mock('cookies-next', () => ({
  getCookie: vi.fn(() => 'light'),
}));

vi.mock('next/headers', () => ({
  cookies: {},
}));

vi.mock('@remix-run/react', async () => {
  const actual = await vi.importActual('@remix-run/react');

  return {
    ...actual,
    useLocation: () => ({ search: '' }),
    useNavigate: () => vi.fn(),
    useParams: () => ({ id: '1' }),
    useSearchParams: () => [{ get: () => '1' }],
  };
});

test('renders NotFound component correctly', () => {
  const { container } = render(<NotFound />);
  expect(container).toMatchSnapshot();
});
