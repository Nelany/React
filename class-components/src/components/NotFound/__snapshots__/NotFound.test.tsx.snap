// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders NotFound component correctly 1`] = `
<div>
  <div
    class="error-content-container light"
  >
    <h1
      class="error-content"
    >
      Ooops... Page not found!
    </h1>
    <img
      alt="Cat"
      class="error-img"
      src="/cat.png"
    />
    <a
      href="/"
    >
      <button
        class="go-to-main-button light"
      >
        Go to Main
      </button>
    </a>
  </div>
</div>
`;
